src/main/java/com/palantir/refreshable/DefaultRefreshable.java

```diff
@@ -190,43 +239,20 @@ public void accept(T value) {
         private final WeakReference<DefaultRefreshable<R>> childRef;
         private final Function<T, R> function;
 
-        @Nullable
-        private Disposable cleanUpSubscription = null;
-
         private SelfRemovingMapSubscriber(Function<T, R> function, DefaultRefreshable<R> child) {
```

Comment:
That could work, however I think the current implementation makes it clear that the instance must still exist when the subscriber is constructed, it can't be garbage collected until it leaves scope (with some caveats around the way the jvm optimizes scope!)

I think there are pros and cons to both, I don't have terribly strong style opinions on these things -- either way we just have to read the code :-)